lib/web_ui/dev/goldens.dart

```diff
@@ -0,0 +1,80 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:image/image.dart';
+
+/// This class encapsulates visually diffing an Image with any other.
+/// Both images need to be the exact same size.
+class ImageDiff {
+
+  /// The image to match
+  final Image golden;
+
+  /// The image being compared
+  final Image other;
+
+  /// The output of the comparison
+  /// Pixels in the output image can have 3 different colors depending on the comparison
+  /// between golden pixels and other pixels:
+  ///  * white: when both pixels are the same
+  ///  * red: when a pixel is found in other, but not in golden
+  ///  * green: when a pixel is found in golden, but not in other
+  Image diff;
+
+  /// The ratio of wrong pixels to all pixels in golden (between 0 and 1)
+  /// This gets set to 1 (100% difference) when golden and other aren't the same size.
+  double get rate => _wrongPixels / _numPixels;
```

Comment:
We talked offline about changing this to `max(pixelColorDeltas(imageA, imageB))`. Did you want to change it in this PR or in a follow-up one?

Reply:
Haha follow up, for now, we have the same features as flutter/flutter (+ tolerances)